cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1362,7 +1362,9 @@ private boolean hasRebalanceAnnotation(KafkaRebalance kafkaRebalance) {
 
     private boolean shouldRenewRebalance(KafkaRebalance kafkaRebalance, KafkaRebalanceAnnotation rebalanceAnnotation) {
         // check if refresh annotation applied or request are made when CC is not active
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))) {
+        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "ConnectException".equals(condition.getReason()))
+                || kafkaRebalance.getStatus().getConditions().stream().anyMatch(condition -> "NoRouteToHostException".equals(condition.getReason()))) {
```

Comment:
It is probably worth adding some comment explaining this. Why does it look for this in the conditions? Also, why does it not use the actual exception classes?

Reply:
As commented before, this might deserve an explanation why this checks for the errors in conditions, because that seems wierd.

Reply:
I agree a better description is needed an that this checking could be made more robust. 

@scholzj what would be a better way to check for these types of exceptions?

Reply:
@tomncooper Well, I do not know why it does what it does -> so it is hard to say what is a _better_ to do it.

But from my PoV ... the controller / operator pattern is based on retrying things in every reconciliation. So why does it need to look into the conditions what the error is? Shouldn't it just know what the error is because it just got the exception from some operation?

Reply:
As I said ... if this is the best way how to deal with something, fine, but it seems unusual at best, so I think it should be explained in the comment. And I also think that this `why does it not use the actual exception classes` is a valid comment even if the logic stays.